bazarr/api/episodes/history.py

```diff
@@ -130,24 +128,36 @@ def get(self):
             'matches': x.matched,
             'dont_matches': x.not_matched,
             'external_subtitles': [y[1] for y in ast.literal_eval(x.external_subtitles) if y[1]],
-            'upgradable': bool(x.upgradable) if _language_still_desired(x.language, x.profileId) else False,
             'blacklisted': bool(x.blacklisted),
         } for x in database.execute(stmt).all()]
 
         for item in episode_history:
-            original_video_path = item['path']
-            original_subtitle_path = item['subtitles_path']
+            still_desired = True
+            if not _language_still_desired(item['language'], item['profileId']):
+                # this language isn't desired anymore so there's no point in upgrading this subtitles
+                still_desired = False
+
             item.update(postprocess(item))
 
-            # Mark not upgradable if score is perfect or if video/subtitles file doesn't exist anymore
+            # Mark upgradeable and get original_id
+            item.update({'original_id': upgradable_episodes_not_perfect.get(item['id'])})
+            language = item['language']['code2']
```

Comment:
I have the same question i asked on the movies here. Where is `language` being used?

Reply:
Same answer ;-)